lib/chef/resource/windows_task.rb

```diff
@@ -94,12 +90,18 @@ def after_created
 
       private
 
+      # Validate the passed value is numeric values only if it is a string
+      def numeric_value_in_string?(val)
+        return false if val.is_a?(String) && /\D/ =~ val # \D is any non-numeric value
```

Comment:
`!!Integer(val) rescue false` (can we convert this thing automatically in to an Integer?)

Reply:
irb(main):002:0> !!Integer(5,000)
=> true
irb(main):003:0> Integer(5,000)
=> 5

This is the problem I was hoping to avoid here. We never should have allowed a String here. Numeric strings are just pain all over

Reply:
But you couldn't pass `5,000` as a bare word anyway:
```
irb(main):004:0> def foo(a); puts a; end
=> :foo
irb(main):005:0> foo 5,000
ArgumentError: wrong number of arguments (given 2, expected 1)
```

Reply:
So here's the problem as it expresses itself in 13.6.4:

```ruby
windows_task 'chef-client' do
  user 'SYSTEM'
  command 'chef-client'
  run_level :highest
  frequency :minute
  frequency_modifier 15
  execution_time_limit '5000'
end
```
The execution_time_limit is properly parsed as being 5000 seconds
<img width="667" alt="correct" src="https://user-images.githubusercontent.com/1015200/33742474-5e5635d0-db5d-11e7-8888-6e56530b7332.png">

```ruby
windows_task 'chef-client' do
  user 'SYSTEM'
  command 'chef-client'
  run_level :highest
  frequency :minute
  frequency_modifier 15
  execution_time_limit '5,000'
end
```

The  execution_time_limit is improperly parsed as being 5 seconds.
<img width="646" alt="incorrect" src="https://user-images.githubusercontent.com/1015200/33742469-574cb67e-db5d-11e7-85af-bd25162f9a86.png">

The real life lesson here is don't set a property to String when you really want an Integer, but that 🚢 is long out of port.

Reply:
Yup:
```
irb(main):001:0> Integer('5,000')
ArgumentError: invalid value for Integer(): "5,000"
```

Reply:
What we see here, really, is that the conversion in `sec_to_dur` on https://github.com/chef/chef/pull/6628/files#diff-7158a2a9329d172ac577a1b48be4ff66R214 isn't right - it just blindy does `to_i`, which tries too hard. If that line were `Integer(seconds)` we'd properly blow up down there and avoid all of this.

Reply:
@tas50 , `execution_time_limit` property was set as string earlier because `schtasks create` accepts input in ISO08601 duration format, which needs to be a string e.g. `PT72H`. Initially this property was accepting input as String in ISO08601 duration format only. It was later changed to accept time in seconds as it's more user friendly.
But yeah the input type should have been changed at that time. But then this [logic](https://github.com/chef/chef/blob/master/lib/chef/provider/windows_task.rb#L339), which fetches all the properties to be updated via task XML in a loop, wouldn't work and there would be more `If-Else` conditions.

Reply:
@thommay I'll just move that into sec_to_dur and rescue the failure there